streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -275,30 +244,69 @@ public void punctuate(ProcessorNode node, long timestamp) {
 
         try {
             node.punctuate(timestamp);
-        } catch (KafkaException ke) {
-            throw new StreamsException(String.format("Exception caught in punctuate. taskId=%s processor=%s", id,  node.name()), ke);
+        } catch (final KafkaException e) {
+            throw new StreamsException(String.format("Exception caught in punctuate. taskId=%s processor=%s", id,  node.name()), e);
         } finally {
             processorContext.setCurrentNode(null);
         }
     }
 
     /**
-     * Commit the current task state
+     * <pre>
+     *  - flush state and producer
+     *  - write checkpoint
+     *  - commit offsets
+     * </pre>
      */
+    @Override
     public void commit() {
-        metrics.metrics.measureLatencyNs(time, commitDelegate, metrics.taskCommitTimeSensor);
+        log.trace("{} Committing", logPrefix);
+        metrics.metrics.measureLatencyNs(
+            time,
+            new Runnable() {
+                @Override
+                public void run() {
+                    flushState();
+                    stateMgr.checkpoint(recordCollectorOffsets());
+                    commitOffsets();
+                }
+            },
+            metrics.taskCommitTimeSensor);
+    }
+
+    /**
+     * <pre>
+     *  - close topology
+     *  - {@link #commit()}
+     *    - flush state and producer
+     *    - write checkpoint
+     *    - commit offsets
+     * </pre>
+     */
+    @Override
+    public void suspend() {
+        log.info("{} Suspending task", logPrefix);
+        closeTopology();
+        log.debug("{} Committing offsets", logPrefix);
```

Comment:
The log level in 263 is `trace` -- but we can remove, as we get `info` log level in 288.